File: 1_two_sum.js
var twoSum = function(nums, target) {
    if (nums.length === 2){
        return [0,1];
    }
    const storage = {};
    storage[nums[0]] = 0;

    for (let i = 1; i < nums.length; i++){
        const difference = target - nums[i];
        if (storage[difference] !== undefined){
            return [storage[difference], i];
        } else {
            storage[nums[i]] = i;
        }
    }
    //Time complexity - O(n), Space complexity - O(1);
};

File: 1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 2:
            return [0,1]
        storage = {}

        for i in range (len(nums)):
            if nums[i] not in storage:
                difference = target - nums[i]
                storage[difference] = i
            else:
                return [storage[nums[i]], i]